egs2/binaural_librispeech/enh1/local/data.sh

```diff
@@ -0,0 +1,168 @@
+    log "stage 0: Downloading BinauralLibriSpeech"
+    # Download and extract BinauralLibriSpeech data
+    if [ -z "${binaural_librispeech}" ]; then
+        if [ -d "${cdir}/downloads/BinauralLibriSpeech" ]; then
+          log "Binaural LibriSpeech data already exists in: ${cdir}/downloads/BinauralLibriSpeech."
+          log "Skipping download.."
+        else
+          mkdir -p "${cdir}/downloads/BinauralLibriSpeech"
+          log "Downloading data from HuggingFace to '${cdir}/downloads/BinauralLibriSpeech'"
+          log "Using Binaural Librispeech version: ${binaural_librispeech_subset}"
+          python local/download_binaural_librispeech.py ${binaural_librispeech_subset} "${cdir}/downloads/"
+          # Remove HF cache after download is finished.
+          rm -rf "${cdir}/downloads/.cache/huggingface/"
+          # Move "BinauralLibriSpeech" up one level to "/downloads"
+          mv "${cdir}/downloads/data/BinauralLibriSpeech" "${cdir}/downloads"
+          # Remove the now-empty "data" folder if desired
+          rmdir "${cdir}/downloads/data"
+          # Extract data
+          log "Extracting data..."
+          local/extract_tar_and_delete.sh "${cdir}/downloads/BinauralLibriSpeech"
+          log "Downloaded and extracted Binaural Librispeech data"
+        fi
+    else
+        # Copy BinauralLibriSpeech to user directory in case of permission issues.
+        log "Binaural LibriSpeech data already exists at ${binaural_librispeech}, copying to '${cdir}/downloads/BinauralLibriSpeech'"
+        mkdir -p "${cdir}/downloads/BinauralLibriSpeech"
+        log "Using Binaural Librispeech version: ${binaural_librispeech_subset}"
+        rsync -r -P "${binaural_librispeech}/${binaural_librispeech_subset}" "${cdir}/downloads/BinauralLibriSpeech"
+    fi
+
+    log "stage 0: Downloading MUSAN noise data to '${cdir}/downloads/musan'"
+    # Download and extract Musan noise data
+    if [ -z "${musan}" ]; then
+        mkdir -p ${cdir}/downloads/musan/
+        num_wavs=$(find "${cdir}/downloads/musan" -iname "*.wav" | wc -l)
+        echo "${num_wavs}"
+        if [ "${num_wavs}" = "930" ]; then
+            echo "'${cdir}/downloads/musan' already exists. Skipping..."
+        else
+            musan_noise_url=https://www.openslr.org/resources/17/musan.tar.gz
+            wget --continue -O "${cdir}/downloads/musan_noise.tar.gz" ${musan_noise_url}
+            tar -zxvf "${cdir}/downloads/musan_noise.tar.gz" -C "${cdir}/downloads/"
+            rm "${cdir}/downloads/musan_noise.tar.gz"
+            rm -rf "${cdir}/downloads/musan/music/"
+            rm -rf "${cdir}/downloads/musan/speech/"
+        fi
+    else
+        log "MUSAN noise data already exists at ${musan}, copying to '${cdir}/downloads/musan'"
+        # Copy Musan to user directory in case of permission issues.
+        mkdir -p "${cdir}/downloads/musan"
+        rsync -r -P "${musan}/noise" "${cdir}/downloads/musan/"
```

Comment:
Is it necessary to copy the downloaded directory? If it is OK to make a symbolic link (via ln -s), it is better to replace it.

Reply:
I guess this depends on the system. Will change to use ls.